phoenix-core/src/main/java/org/apache/phoenix/schema/PMetaDataImpl.java

```diff
@@ -160,7 +160,7 @@ public PMetaDataCache cloneMinusOverage(long overage) {
                     newCache.put(tableRef.table.getKey(), new PTableRef(tableRef));
                     toRemove.add(tableRef);
                     toRemoveBytes += tableRef.estSize;
-                    if (toRemoveBytes - toRemove.peekLast().estSize > overage) {
+                    while (toRemoveBytes - toRemove.peekLast().estSize >= overage) {
```

Comment:
I've added a couple more tests.

The `cloneMinusOverage()` method does sometimes need to clear everything out, because it's what happens prior to adding the new thing, so it's correct for it not to break out of the loop when the size is one. I've added a test to confirm this (`shouldAlwaysKeepAtLeastOneEntryEvenIfTooLarge()`).

I've also added a test to check things work out OK when the max size is zero. In this case there should always be exactly one thing in the cache (after the first item has been added). This test also passes.